maint-lib/stagelib/filetools.py

```diff
@@ -70,6 +71,12 @@ def copy_files(filenames, src_path, dst_path, files_copied={}):
         logging.info(COPY_LOGTEXT.format(dirty_marker + ' ' + file_path, dst_path))
         files_copied[os.path.join(dst_path, f)] = dirty_marker + ' ' + file_path
         _copy_file(file_path, dst_path)
+        if f.endswith("_PACKAGES__"):
```

Comment:
All of this code is interleaved into existing code. It make it harder to read/modify all of the code later. We should do this in a separate step. I think we copy files into staging as normal, and then in staging, look for files matching `_PACKAGE` (to support `__RADOSGW_PACKAGE__`, for exaple) or `_PACKAGES`, and then make a copy of it there named `_UNVERSIONED_PACKAGE` or `_UNVERSIONED_PACKAGES` without the `X_POINT_RELEASE` env replacment. We also need to replace any reference to `_PACKAGES` or `_PACKAGE` with `_UNVERSIONED_PACKAGE` or `_UNVERSIONED_PACKAGES` respectively.

This way the normal file replacement doesn't have to change, and all of this code can be viewed and edited separately from the rest.